packages/sdk/contractkit/src/wrappers/MultiSig.ts

```diff
@@ -81,31 +83,59 @@ export class MultiSigWrapper extends BaseWrapper<MultiSig> {
   ) {
     const data = stringToSolidityBytes(txo.encodeABI())
     const transactionCount = await this.getTransactionCount(true, true)
-    // reverse order for recency
-    for (let transactionId = transactionCount - 1; transactionId >= 0; transactionId--) {
-      const tx = await this.getTransaction(transactionId)
-      if (tx.data === data && tx.destination === destination && tx.value.isEqualTo(value)) {
-        return tx
-      }
+    const transactionIndexes = await Promise.all(
+      Array(transactionCount)
+        .fill(0)
+        .map(async (_, index) => {
+          const tx = await this.getTransaction(index, false)
+          if (tx.data === data && tx.destination === destination && tx.value.isEqualTo(value)) {
+            return index
+          }
+          return null
+        })
+    )
+    const txID = transactionIndexes.find((index) => index !== null)
+    if (!txID) {
+      return
```

Comment:
Aren't we overfetching now?

Reply:
true, would be better to add a way to just fetch confirmations for a transaction